components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/EnvironmentSecretRepository.java

```
@@ -0,0 +1,80 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.micro.integrator.mediation.security.vault;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.SynapseException;
+import org.wso2.securevault.DecryptionProvider;
+import org.wso2.securevault.secret.SecretRepository;
+
+import java.util.Properties;
+
+public class EnvironmentSecretRepository implements SecretRepository {
+
+    private static Log LOG = LogFactory.getLog(EnvironmentSecretRepository.class);
+
+    /* Parent secret repository */
+    private SecretRepository parentRepository;
+
+    @Override
+    public void init(Properties properties, String s) {
+        // Nothing to do here
+    }
+
+    @Override
+    public String getSecret(String alias) {
+        String rawValue = getPlainTextSecret(alias);
+        DecryptionProvider decyptProvider = CipherInitializer.getInstance().getDecryptionProvider();
+        if (decyptProvider == null) {
+            // This cannot happen unless someone mess with OSGI references
+            LOG.error("Can not proceed decryption due to the secret repository initialization error");
+            throw new SynapseException("Secret repository is null for alias " + alias);
+        }
+        return new String(decyptProvider.decrypt(rawValue.trim().getBytes()));
+    }
+
+    /**
+     * Returns the plaintext value from the environment.
+     *
+     * @param alias Environment variable name.
+     */
+    public String getPlainTextSecret(String alias) {
+        String planTextSecret = System.getenv(alias);
+        if (planTextSecret == null || planTextSecret.isEmpty()) {
+            throw new SynapseException("Plain text secret value has not been set for variable " + alias);
```

Should we throw an exception here  or just return whatever comes as null or empty. The user might write the logic based on the availability of multiple env properties.

---

+1. done